src/repr/src/fixed_length.rs

```diff
@@ -0,0 +1,752 @@
+}
+
+// TODO(vmarcos): We may be able to simplify the implementation of Columnation
+// once frankmcsherry/columnation#18 lands.
+impl<const N: usize> Columnation for BytesN<N> {
+    type InnerRegion = CopyRegion<BytesN<N>>;
+}
+
+/// Helper function to read 8 bits as `bool`s out of a `u8`.
+fn u8_to_bool_slice(b: u8) -> [bool; 8] {
+    let mut result = [false; 8];
+    for i in 0..8 {
+        result[i] = (b & (1u8 << i)) != 0u8;
+    }
+    result
+}
+
+/// Helper function to obtain the fixed length of a `ScalarType`, if there is one.
+pub(super) fn fixed_length(typ: &ScalarType) -> Option<usize> {
+    match typ {
+        ScalarType::Bool => Some(size_of::<u8>()),
+        ScalarType::Int16 => Some(size_of::<i16>()),
+        ScalarType::Int32 => Some(size_of::<i32>()),
+        ScalarType::Int64 => Some(size_of::<i64>()),
+        ScalarType::UInt16 => Some(size_of::<u16>()),
+        ScalarType::UInt32 => Some(size_of::<u32>()),
+        ScalarType::UInt64 => Some(size_of::<u64>()),
+        ScalarType::Float32 => Some(size_of::<f32>()),
+        ScalarType::Float64 => Some(size_of::<f64>()),
+        ScalarType::Date => Some(size_of::<i32>()),
+        ScalarType::Time => Some(2 * size_of::<u32>()),
+        ScalarType::Timestamp { precision: _ } => Some(3 * size_of::<i32>() + size_of::<u32>()),
+        ScalarType::TimestampTz { precision: _ } => Some(3 * size_of::<i32>() + size_of::<u32>()),
+        ScalarType::Interval => Some(2 * size_of::<i32>() + size_of::<i64>()),
+        ScalarType::Uuid => Some(size_of::<uuid::Bytes>()),
+        ScalarType::Oid => Some(size_of::<u32>()),
+        ScalarType::MzTimestamp => Some(size_of::<Timestamp>()),
+        ScalarType::MzAclItem => Some(MzAclItem::binary_size()),
+        ScalarType::AclItem => Some(AclItem::binary_size()),
+        _ => None,
+    }
+}
+
+impl<'a, const N: usize> IntoRowByTypes<'a> for BytesN<N> {
+    /// This implementation employs the schema provided by `types` to decode
+    /// the N-byte value into `row_buf`, returning a reference to it.
+    ///
+    /// Presently, the implementation panics if more than one `ColumnType` is found
+    /// in `types` or if the one column has a byte length greater than N-1 (due to one
+    /// byte being reserved for nullability encoding).
+    fn into_row(&'a self, row_buf: &'a mut Row, types: &[ColumnType]) -> &'a Row {
+        // TODO (vmarcos): The implementation should be extended in the future to support
+        // multiple columns in `types` and to panic if the multi-column schema that ends up
+        // being too wide for a fixed-length N-byte pattern (MaterializeInc/materialize#22102).
+        assert!(N > 1);
+        assert!(types.len() == 1);
+        let typ = &types[0];
+        let mut packer = row_buf.packer();
+        let mut offset = 0;
+
+        // Firstly, read the nullability indicators.
+        let nullability =
+            u8_to_bool_slice(u8::from_le_bytes(read_byte_array(&self.data, &mut offset)));
```

Comment:
The reasoning here was to keep the code as close to what is done in `Row` as possible, so that we do not have two different ways to decode/encode basic data (and two different performance debugging problems as a consequence!).